secure_message/services/case_service.py

```diff
@@ -11,16 +11,21 @@ class CaseService:
     def store_case_event(case_id, user_name):
         """posts the data to the case service"""
 
-        json_data = {"description": "New Secure Message",
-                     "category": "SECURE_MESSAGE_SENT",
-                     "createdBy": user_name}
+        if case_id:
+            json_data = {"description": "New Secure Message",
+                         "category": "SECURE_MESSAGE_SENT",
+                         "createdBy": user_name}
 
-        url = current_app.config['RM_CASE_POST'].format(current_app.config['RM_CASE_SERVICE'], case_id)
-        case_service_data = requests.post(url, auth=current_app.config['BASIC_AUTH'], json=json_data, verify=False)
-        logger.debug('case service post result', status_code=case_service_data.status_code,
-                     reason=case_service_data.reason, text=case_service_data.text)
-        case_service_dict = json.loads(case_service_data.text)
-        if 'error' in case_service_dict:
-            case_service_dict = case_service_dict['error']
+            url = current_app.config['RM_CASE_POST'].format(current_app.config['RM_CASE_SERVICE'], case_id)
+            case_service_data = requests.post(url, auth=current_app.config['BASIC_AUTH'], json=json_data, verify=False)
+            logger.debug('case service post result', status_code=case_service_data.status_code,
+                         reason=case_service_data.reason, text=case_service_data.text)
+            case_service_dict = json.loads(case_service_data.text)
+            if 'error' in case_service_dict:
+                case_service_dict = case_service_dict['error']
 
-        return case_service_dict, case_service_data.status_code
+            return case_service_dict, case_service_data.status_code
+        else:
+            err = f"No case id for case involving user {user_name}, case event not called"
+            logger.error(err)
```

Comment:
🎨 Could have this in one line

Reply:
no because it logs the error then returns it